src/compas/datastructures/tree/tree.py

```diff
@@ -450,22 +449,27 @@ def print_hierarchy(self, max_depth=None):
 
         Returns
         -------
-        None
+        str
+            The spatial hierarchy of the tree.
 
         """
 
-        def _print(node, prefix="", last=True, depth=0):
+        hierarchy = {"string": ""}
+
+        def traverse(node, hierarchy, prefix="", last=True, depth=0):
 
             if max_depth is not None and depth > max_depth:
                 return
 
             connector = "└── " if last else "├── "
-            print("{}{}{}".format(prefix, connector, node))
+            hierarchy["string"] += "{}{}{}\n".format(prefix, connector, node)
```

Comment:
why do you pass around a dictionary with a single, hard-coded key?

Reply:
This was to make the string a mutable object (like a pointer). In python 3 i can simply use `nonlocal` and just say `hierarchy_string`, but that option is not available for 2.7. Maybe now looks better with the list?